docs/docs/reference/built-in-components/gatsby-script.md

```diff
@@ -0,0 +1,282 @@
+
+## Migrating existing scripts
+
+For most scripts, you can migrate to Gatsby Script by importing the `<Script>` component in your file and changing the lowercase `script` tag names to capitalized `Script` tag names:
+
+```diff
+import { Script } from "gatsby";
+
+-<script src="https://my-example-script" />
++<Script src="https://my-example-script" />
+```
+
+By default, the `<Script>` component will load your script after hydration. For more information on declaring loading strategies, see the [Strategies](#strategies) section.
+
+## Scripts with sources and inline scripts
+
+There are two types of scripts that you can tell the `<Script>` component to load:
+
+- Scripts with sources
+- Inline scripts
+
+### Scripts with sources
+
+Scripts with sources provide a `src` property like this:
+
+```jsx
+<Script src="https://my-example-script" />
+```
+
+The `<Script>` component will use the value of `src` to deduplicate loading, so if you include two scripts on the same page with the same `src`, only one will load.
+
+If for some reason you need to load two scripts with the same sources on the same page, you can provide an optional, unique `id` property to each and the `<Script>` component will attempt to load both:
+
+```jsx
+<Script id="first-unique-id" src="https://my-example-script" />
+<Script id="second-unique-id" src="https://my-example-script" />
+```
+
+### Inline scripts
+
+Inline scripts must include a unique `id` property and can be defined in two ways:
+
+- Via React's special [`dangerouslySetInnerHTML`](https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml) property
+- Via a template literal
+
+Here's a look at both:
+
+```jsx
+<Script id="first-unique-id" dangerouslySetInnerHTML={{ __html: `alert('Hello world')` }} />
+<Script id="second-unique-id">{`alert('Hello world')`}</Script>
+```
+
+Functionally, both of these ways of defining inline scripts are equivalent.
+
+## Strategies
+
+You can declare a loading strategy by passing a `strategy` property. These are the available loading strategies:
+
+- `post-hydrate` - Loads after the page has hydrated
+- `idle` - Loads after the page has become idle
+- `off-main-thread` - Loads off the main thread in a web worker via [Partytown](https://partytown.builder.io)
+
+The best strategy to use depends on the functionality of the script you would like to include.
+
+A general rule of thumb is if your script can afford to be loaded later (as many analytics scripts can be), then start with `off-main-thread` or `idle` and move to earlier loading strategies if necessary.
+
+Here's how you can define these strategies in the `<Script>` component:
+
+```jsx
+<Script src="https://my-example-script" strategy="post-hydrate" />
+<Script src="https://my-example-script" strategy="idle" />
+<Script src="https://my-example-script" strategy="off-main-thread" />
+```
+
+Additionally, Gatsby exports a `ScriptStrategy` enum that you can use in TSX files if you prefer:
```

Comment:
Do we need to state the 2 approaches here? I would recommend showing only the `ScriptStrategy` as it is less error-prone but I don't have strong feelings about it

Reply:
JavaScript users will get code completion for string literals so I think it's useful to document both. It was also a [point of feedback in the RFC](https://github.com/gatsbyjs/gatsby/discussions/35404#discussioncomment-2614320)